spark/src/test/scala/org/apache/comet/parquet/ParquetReadSuite.scala

```diff
@@ -1946,6 +1946,52 @@ class ParquetReadV1Suite extends ParquetReadSuite with AdaptiveSparkPlanHelper {
       }
     }
   }
+
+  test("read basic complex types") {
+    Seq(true, false).foreach(dictionaryEnabled => {
+      withTempPath { dir =>
+        val path = new Path(dir.toURI.toString, "complex_types.parquet")
+        makeParquetFileComplexTypes(path, dictionaryEnabled, 10)
+        withParquetTable(path.toUri.toString, "complex_types") {
+          Seq(CometConf.SCAN_NATIVE_DATAFUSION, CometConf.SCAN_NATIVE_ICEBERG_COMPAT).foreach(
+            scanMode => {
+              withSQLConf(CometConf.COMET_NATIVE_SCAN_IMPL.key -> scanMode) {
+                checkSparkAnswer(sql("select * from complex_types"))
```

Comment:
What about using checkSparkAnswerAndOperator in order to make sure it is not falling back?

Reply:
Done. Had to modify the query to not use `map_entries` (which is not implemented at this time)